src/pages/NovoColetorScreen.jsx

```diff
@@ -0,0 +1,167 @@
+import { Component } from 'react'
+
+import {
+
+    Row,
+    Col,
+    Divider,
+    Input,
+    Button,
+    notification,
+    Spin
+} from 'antd'
+import axios from 'axios'
+
+import { Form } from '@ant-design/compatible'
+
+const FormItem = Form.Item
+
+class NovoColetorScreen extends Component {
+    constructor(props) {
+        super(props)
+        this.state = { loading: false }
+    }
+
+    async componentDidMount() {
+        if (this.props.match.params.coletor_id !== undefined) {
+            await this.buscarColetorPorId()
+        }
+    }
+
+    notificacao = (type, message, description) => {
+        notification[type]({
+            message,
+            description
+        })
+    }
+
+    handleSubmit = (err, valores) => {
+        if (!err) {
+            if (this.props.match.params.coletor_id !== undefined) {
+                this.requisitaEdicaoColetor(valores)
+            } else {
+                this.requisitaCadastroColetor(valores)
+            }
+        }
+    }
+
+    onSubmit = event => {
+        event.preventDefault()
+        this.props.form.validateFields(this.handleSubmit)
+    }
+
+    requisitaCadastroColetor = async valores => {
+        this.setState({ loading: true })
+
+        try {
+            await axios.post('/coletores', valores)
+
+            this.notificacao('success', 'Sucesso', 'Coletor cadastrado com sucesso.')
+
+            this.props.history.goBack()
```

Comment:
Faz o redirecionamento para a listagem de coletores de forma explícita. Isso evita casos em que eu acesse a URL diretamente pelo browser, uma vez que você faz um `goBack()` o navegador vai voltar para a página anterior do usuário, seja lá qual for. Faz o mesmo comportamento das outras tela de cadastro que você criou. Não precisa mudar o que já existia.